addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,505 @@
+                description: _t("Hide Text under a Toggle"),
+                icon: "fa-caret-square-o-right",
+                isAvailable: (node) => !closestElement(node, titleSelector),
+                run: () => {
+                    this.insertToggleList();
+                },
+            },
+        ],
+        powerbox_items: [
+            {
+                commandId: "insertToggleList",
+                categoryId: "structure",
+                title: _t("Toggle List"),
+                description: _t("Hide Text under a Toggle"),
+            },
+        ],
+        normalize_handlers: this.normalize.bind(this),
+        delete_backward_overrides: this.handleDeleteBackward.bind(this),
+        delete_forward_overrides: this.handleDeleteForward.bind(this),
+        tab_overrides: this.handleTab.bind(this),
+        shift_tab_overrides: this.handleShiftTab.bind(this),
+        hints: [
+            withSequence(1, {
+                selector: `${titleSelector} > p`,
+                text: "Toggle Title",
+            }),
+            withSequence(2, {
+                selector: `${contentSelector} > p:only-child`,
+                text: "This is empty, add some text",
+            }),
+        ],
+        split_element_block_overrides: this.handleSplitElementBlock.bind(this),
+        power_buttons_visibility_predicates: this.showPowerButtons.bind(this),
+        before_paste_handlers: this.beforePaste.bind(this),
+        disallowed_list_selectors: `${titleSelector} *, div.collapsed > [data-embedded-editable='content'] *`,
+        mount_component_handlers: this.setupNewToggle.bind(this),
+        select_all_handlers: this.selectAll.bind(this),
+    };
+
+    selectAll() {
+        const selection = this.dependencies.selection.getEditableSelection();
+        if (closestElement(selection.anchorNode, "[data-embedded='toggle']")) {
+            const node = findFurthest(
+                selection.anchorNode,
+                this.editable,
+                (element) => element.dataset.embedded === "toggle"
+            );
+            if (node) {
+                this.dependencies.selection.setSelection({
+                    anchorNode: node?.previousSibling || node,
+                    anchorOffset: 0,
+                });
+            }
+        }
+    }
+
+    setup() {
+        const allToggles = this.editable.querySelectorAll("[data-embedded='toggle']");
+        allToggles.forEach((toggleNode) => {
+            this.addDomListener(toggleNode, "keydown", (ev) => {
+                if (["arrowup", "arrowdown"].includes(getActiveHotkey(ev))) {
+                    this.handleKeyDown(ev);
+                }
+            });
+        });
+    }
+
+    handleKeyDown(ev) {
+        const selection = this.dependencies.selection.getEditableSelection();
+        const targetEl = closestElement(selection.anchorNode, isBlock);
+        if (targetEl.previousSibling && targetEl.nextSibling) {
+            return;
+        }
```

Comment:
I tried changing this to use the modifySelection method but I encountered the same issue as I explained to you last week.
Since we are inside of a different `contenteditable=true` element, we can only move the selection at the same level.
Do you remember how you managed to use it so that you have the same behavior?